src/robotFactory.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
+/* eslint max-len: ["error", { "ignoreTemplateLiterals": true }] */
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = Object.keys(coords).length ? coords
+      : {
+        x: 0, y: 0,
+      };
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = this.coords.z === undefined ? 0 : coords.z;
```

Comment:
You can simplify this statement: 
```suggestion
    this.coords.z = coords.z || 0;
```